app/controllers/provider_interface/candidate_pool/provider_invite_messages_controller.rb

```diff
@@ -0,0 +1,84 @@
+module ProviderInterface
+  module CandidatePool
+    class ProviderInviteMessagesController < ProviderInterfaceController
+      before_action :set_candidate
+      before_action :set_back_path, only: %i[edit]
+
+      def new
+        @pool_invite = PoolInviteMessageForm.new(invite:)
+        @course = invite.course
+      end
+
+      def edit
+        @pool_invite = PoolInviteMessageForm.new(
+          invite:,
+          invite_message_params: {
+            provider_message: invite.provider_message,
+            message_content: invite.message_content,
+          },
+        )
+        @course = invite.course
+      end
+
+      def create
+        @pool_invite = PoolInviteMessageForm.new(
+          invite:,
+          invite_message_params:,
+        )
+
+        if @pool_invite.valid?
+          @pool_invite.save
+          redirect_to provider_interface_candidate_pool_candidate_draft_invite_path(@candidate, invite)
+        else
+          @course = invite.course
+          render :new
+        end
+      end
+
+      def update
+        @pool_invite = PoolInviteMessageForm.new(
+          invite:,
+          invite_message_params:,
+        )
+
+        if @pool_invite.valid?
+          @pool_invite.save
+          redirect_to provider_interface_candidate_pool_candidate_draft_invite_path(@candidate, invite)
+        else
+          @course = pool_invite.course
+          render :new
+        end
+      end
+
+    private
+
+      def set_candidate
+        @candidate ||= Pool::Candidates.application_forms_for_provider
+          .find_by(candidate_id: params.expect(:candidate_id)).candidate
+      end
+
+      def set_back_path
+        if params[:return_to] == 'review'
+          @back_path = provider_interface_candidate_pool_candidate_draft_invite_path(
+            @candidate,
+            invite,
+          )
+        end
+      end
+
+      def invite
+        @invite ||= Pool::Invite.find_by(
+          id: params.expect(:draft_invite_id),
+          provider_id: current_provider_user.provider_ids,
+          status: :draft,
+        )
```

Comment:
```suggestion
        @invite ||= current_provider_user.providers_invites.draft.find(params.expect(:draft_invite_id))
```

Could we add a `providers_invites` relationship on the ProviderUser which can get the Invites scoped to their Providers?
Changing this would allow us to remove the Brakeman warning